modules/backend/clinic/staff/Staff.cpp

```diff
@@ -1,18 +1,60 @@
 #include "Staff.hpp"
+#include <algorithm>
+#include <memory>
 
 namespace clinic
 {
 namespace staff
 {
 
-void Staff::addDoctor(const doctor::Doctor& doctor)
+template <typename Person> typename Person::iterator Staff::getPersonByName(Person& person, const common::Name& name)
 {
-    doctors_.emplace_back(doctor);
+    return std::find_if(person.begin(), person.end(), [&name](const auto& per) {
+        return per->GetPersonalData().getName().first_name_ == name.first_name_ &&
+               per->GetPersonalData().getName().last_name_ == name.last_name_;
```

Comment:
why call GetPersonalData() twice?
I suggest storing in local variable and then querry for first_name or last_name.

or something like
```
const auto& [first_name, last_name] = per->GetPersonalData();
return ...
```